docs/DriverParseableOutput.rst

```diff
@@ -33,6 +33,25 @@ This allows the driver to pass as much information as it wants about the ongoing
 compilation, and programs which parse this data can decide what to use and what
 to ignore.
 
+
+Quasi-PIDs
+==========
+
+In previous versions of this format, certain fields labeled "pid" carry only
+legitimate process IDs (PIDs) from the underlying operating system. Real PIDs
+are always positive numbers, in keeping with POSIX semantics and the unsigned
+Process ID type on Windows.
+
+In more recent versions of this format, the Swift driver combines multiple jobs
+together into batches, running each batch in a single subprocess. In this mode,
+the driver writes messages describing each job as before, but assigns to each a
+so-called "quasi-PID" for the "pid" field, which does not correspond to an
+operating system process ID. Rather, a quasi-PID is a **negative number** that
+can be used **like** a PID, both as a unique identifier for each job that was
+run and in order to relate messages and tasks together, but it does not denote a
+real process.
```

Comment:
We choose them because the current semantics of process IDs on all platforms is that they're positive numbers; negative numbers are unused as identifiers, only as error codes. We are not emitting error codes as identifiers here (indeed, if we did it would break assumptions in the client that the "pid" field is an identifier -- two tasks could have the same error code).

I don't really know what you want by "more" here: more persuasion in the documentation that this is a future-safe design? I can't say that it is for sure, there are always unforeseen future problems, but the existing users of the "pid" only use it as an identifier, and we're preserving the ability to do that, just not as an OS-meaningful-identifier-for-a-process.

Reply:
I'll note again to not mention Xcode unless something really is specific to Xcode. The "pid" field in parseable output is the only way to match up "began" messages with "finished" messages.

Reply:
Included these sentences (minus Xcode mention) in update.

Reply:
Thanks!